### ISSUE TITLE
Specs fail with ruby 2.2.3 (2.2.x?)

### ISSUE THREAD

**Issue by jhart-r7** [rapid7/metasploit-framework#6015]
With Ruby 2.2.3 managed with `rbenv` on Ubuntu 12.04, after `bundle install`, both `bundle exec rspec` and `rake spec` fail with:

```
/home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:229:in `require': cannot load such file -- test/unit/assertions (LoadError)
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:229:in `block in require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:214:in `load_dependency'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:229:in `require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-rails-2.99.0/lib/rspec/rails/adapters.rb:17:in `rescue in <module:Rails>'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-rails-2.99.0/lib/rspec/rails/adapters.rb:12:in `<module:Rails>'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-rails-2.99.0/lib/rspec/rails/adapters.rb:6:in `<module:RSpec>'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-rails-2.99.0/lib/rspec/rails/adapters.rb:5:in `<top (required)>'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:229:in `require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:229:in `block in require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:214:in `load_dependency'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.0.13/lib/active_support/dependencies.rb:229:in `require'
	from /home/jhart/rapid7/metasploit-framework/spec/spec_helper.rb:19:in `<top (required)>'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1036:in `require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1036:in `block in setup_load_path_and_require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1036:in `each'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/configuration.rb:1036:in `setup_load_path_and_require'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/configuration_options.rb:25:in `configure'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/command_line.rb:17:in `run'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/runner.rb:103:in `run'
	from /home/jhart/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-2.99.2/lib/rspec/core/runner.rb:17:in `block in autorun'
```

Switching to the officially supported 2.1.6 fixes this.

**Comment by Dima12345** [comment]
I have this problem too, but I have rails 2.1.5

**Comment by bcook-r7** [comment]
@limhoff-r7 has done a swag at fixing this. I understand that we may need to update rspec itself for Ruby 2.2

**Comment by limhoff-r7** [comment]
RSpec 3.x is compatible with Ruby 2.2, so I'm working on upgrading framework to 2.2: https://github.com/limhoff-r7/metasploit-framework/tree/chore/MSP-13484/rspec-3

**Comment by jhart-r7** [comment]
https://github.com/rapid7/metasploit-framework/pull/6326 fixes part of this and the specs now run, but many still fail, like:

```
Metasploit::Framework::LoginScanner::VMAUTHD ..............................................F.......

  2) Metasploit::Framework::LoginScanner::VMAUTHD behaves like Metasploit::Framework::LoginScanner::Base validations host is not valid for DNS name that cannot be resolved
     Failure/Error: expect(login_scanner.errors[:host]).to include "could not be resolved"
       expected [] to include "could not be resolved"
     Shared Example Group: "Metasploit::Framework::LoginScanner::Base" called from ./spec/lib/metasploit/framework/login_scanner/vmauthd_spec.rb
```

**Comment by bcook-r7** [comment]
We are on ruby 2.3.1 now, a lot of specs fixed with the Rails 4.2 merge.
